nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/GetHDFSFileInfo.java

```diff
@@ -266,6 +268,31 @@ public void onPropertyModified(PropertyDescriptor descriptor, String oldValue, S
         req = null;
     }
 
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        final List<ValidationResult> validationResults = new ArrayList<>(super.customValidate(validationContext));
+
+        String destination = validationContext.getProperty(DESTINATION).getValue();
+        String grouping = validationContext.getProperty(GROUPING).getValue();
+        String batchSize = validationContext.getProperty(BATCH_SIZE).getValue();
+
+        if (
+            (!DESTINATION_CONTENT.getValue().equals(destination) || !GROUP_NONE.getValue().equals(grouping))
+                && batchSize != null
+                && !batchSize.equals("")
```

Comment:
StringUtils.isEmpty() could be used to simplify the condition.
We have a StringUtils in NiFi and commons-lang3's StringUtils is also used. The latter is already imported here, so I'd use that one (but I don't know if there is a convention for it).